boilerplate/modules/DevtoolsServer.jsm

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use strict";
+
+const { useDistinctSystemPrincipalLoader } = ChromeUtils.import(
+  "resource://devtools/shared/loader/Loader.jsm"
+);
+
+var EXPORTED_SYMBOLS = ["DevtoolsServer"];
+
+/**
+ * Opens a devtool server that can be inspected using Firefox's devtools.
+ *
+ * The port can be set either by passing it in to the function or setting the
+ * appropriate preference value. If there is no port value provided, Gecko will
+ * find a free port and use it instead
+ *
+ * To connect to the devtools server:
+ * 1. Open 'about:debugging' in Firefox or Pulse Browser
+ * 2. Open the setup tab
+ * 3. Enter the following host inside of network locations 'localhost:${listener.port}' and click 'Add'
+ * 4. Click 'Connect' in the left sidebar
+ * 5. Click on the entry in the left sidebar
+ * 6. Click 'Open in Browser Toolbox'
+ */
+class DevtoolsServer {
+  /**
+   * @param {?number} port The port you want to open the devtools on.
```

Comment:
[The documentation's example](https://jsdoc.app/tags-type.html) has the `?` before the type, so I am not so sure. Although it is probably more correct to be put as an optional type rather than a nullable type, so I will change that.